orchagent/portsorch.cpp

```diff
@@ -1704,22 +1704,24 @@ void PortsOrch::doPortTask(Consumer &consumer)
                             continue;
                         }
 
-                        if (getPortSpeed(p.m_port_id, current_speed))
+                        if (!getPortSpeed(p.m_port_id, current_speed))
                         {
-                            if (speed != current_speed)
+                            SWSS_LOG_ERROR("Failed to get current speed for port %s", alias.c_str());
+                            it++;
+                            continue;
+                        }
+
+                        if (speed != current_speed)
+                        {
+                            bool old_admin_state_up = (p.m_admin_state == true);
+
+                            /* if port is up, set it down before applying speed */
+                            if (old_admin_state_up)
                             {
                                 if (setPortAdminStatus(p.m_port_id, false))
                                 {
-                                    if (setPortSpeed(p.m_port_id, speed))
-                                    {
-                                        SWSS_LOG_NOTICE("Set port %s speed to %u", alias.c_str(), speed);
-                                    }
-                                    else
-                                    {
-                                        SWSS_LOG_ERROR("Failed to set port %s speed to %u", alias.c_str(), speed);
-                                        it++;
-                                        continue;
-                                    }
+                                    p.m_admin_state = false;
```

Comment:
Do we need to set this and copy ` p `and reset it later below since this is temporary status change?